packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2410,7 +2410,11 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
       modelColumn,
       addToExisting
     );
-    this.updateSorts(sorts);
+    if (model.columns[columnIndex].isSortable ?? false) {
```

Comment:
The `const sorts = TableUtils.toggleSort ...` should be inside this if block, before the `updateSorts`.. no point creating the sort if it's not sortable.
Also should be checking `model.isColumnSortable` here.